### ISSUE TITLE
Problems loading with many dependencies

### ISSUE THREAD

**Issue by josh-degraw** [vscode-contrib/vscode-versionlens#131]
I've looked up and couldn't see any issues that seemed to match what I'm talking about, but if there is one already go ahead and remove this.

I have a couple projects that I work on and with the (relatively) smaller ones this works well. But I have another project with a lot of dependencies and the extension just doesn't load. The spinner runs for several minutes and if I'm lucky it will load, but often it doesn't ever load, or if it does, it takes longer than 10 minutes and I just give up.

This extension has been invaluable for updating my npm dependencies, but when it doesn't load it's basically useless now.

**Comment by MatBdry** [comment]
I got the same issue since the version 1.13.1 of vscode. It's running infinitely...

**Comment by tufantunc** [comment]
I've got the same issue after the update version 1.31.1.

**Comment by joelmuskwe** [comment]
same issue too

**Comment by pflannery** [comment]
Does anyone have an example I can work with?

**Comment by joelmuskwe** [comment]
mine is now working but example of the package.json that was giving problems is as follows:
`{
  "name": "project-name",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "./node_modules/.bin/nx lint && ng lint",
    "e2e": "ng e2e",
    "affected:apps": "./node_modules/.bin/nx affected:apps",
    "affected:libs": "./node_modules/.bin/nx affected:libs",
    "affected:build": "./node_modules/.bin/nx affected:build",
    "affected:e2e": "./node_modules/.bin/nx affected:e2e",
    "affected:test": "./node_modules/.bin/nx affected:test",
    "affected:lint": "./node_modules/.bin/nx affected:lint",
    "affected:dep-graph": "./node_modules/.bin/nx affected:dep-graph",
    "format": "./node_modules/.bin/nx format:write",
    "format:write": "./node_modules/.bin/nx format:write",
    "format:check": "./node_modules/.bin/nx format:check",
    "update": "ng update @nrwl/schematics",
    "update:check": "ng update",
    "workspace-schematic": "./node_modules/.bin/nx workspace-schematic",
    "dep-graph": "./node_modules/.bin/nx dep-graph",
    "help": "./node_modules/.bin/nx help",
    "affected": "./node_modules/.bin/nx affected"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^7.0.0",
    "@angular/common": "^7.0.0",
    "@angular/compiler": "^7.0.0",
    "@angular/core": "^7.0.0",
    "@angular/forms": "^7.0.0",
    "@angular/platform-browser": "^7.0.0",
    "@angular/platform-browser-dynamic": "^7.0.0",
    "@angular/router": "^7.0.0",
    "@auth0/angular-jwt": "^2.1.0",
    "@coreui/angular": "^2.2.4",
    "@coreui/coreui": "^2.1.6",
    "@coreui/coreui-plugin-chartjs-custom-tooltips": "^1.2.0",
    "@coreui/icons": "^0.3.0",
    "@ng-idle/core": "^6.0.0-beta.3",
    "@ng-idle/keepalive": "^6.0.0-beta.3",
    "@ngrx/effects": "7.2.0",
    "@ngrx/entity": "6.1.0",
    "@ngrx/router-store": "7.2.0",
    "@ngrx/store": "7.2.0",
    "@ngx-translate/core": "^11.0.1",
    "@ngx-translate/http-loader": "^4.0.0",
    "@nrwl/nx": "7.6.2",
    "@sweetalert2/ngx-sweetalert2": "^5.0.1",
    "angular-pipes": "^9.0.2",
    "angular2-moment": "^1.9.0",
    "angular2-toaster": "^7.0.0",
    "bootstrap": "^4.3.1",
    "chart.js": "^2.8.0",
    "core-js": "^2.5.4",
    "flag-icon-css": "^3.3.0",
    "font-awesome": "^4.7.0",
    "google-libphonenumber": "^3.2.2",
    "libphonenumber-js": "^1.7.15",
    "moment": "^2.24.0",
    "mutationobserver-shim": "^0.3.3",
    "ng-bootstrap-form-validation": "^4.0.0",
    "ng-spin-kit": "^5.1.1",
    "ng2-charts": "^2.2.2",
    "ngx-bootstrap": "^4.0.1",
    "ngx-loading": "^3.0.1",
    "ngx-pagination": "^3.2.1",
    "ngx-perfect-scrollbar": "^7.2.0",
    "ngx-permissions": "^6.0.5",
    "ngx-uploader": "^7.1.0",
    "rxjs": "~6.4.0",
    "simple-line-icons": "^2.4.1",
    "sweetalert2": "^8.8.7",
    "tslib": "^1.9.0",
    "v8n": "^1.2.3",
    "zone.js": "^0.8.26"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.13.1",
    "@angular/cli": "7.3.1",
    "@angular/compiler-cli": "^7.0.0",
    "@angular/language-service": "^7.0.0",
    "@ngrx/schematics": "7.2.0",
    "@ngrx/store-devtools": "7.2.0",
    "@nrwl/builders": "7.6.2",
    "@nrwl/schematics": "7.6.2",
    "@types/jasmine": "~2.8.6",
    "@types/jasminewd2": "~2.0.3",
    "@types/jest": "^23.0.0",
    "@types/jquery": "3.3.6",
    "@types/node": "~8.9.4",
    "codelyzer": "~4.5.0",
    "cypress": "^3.1.0",
    "dotenv": "6.2.0",
    "jasmine-core": "~2.99.1",
    "jasmine-marbles": "0.4.0",
    "jasmine-spec-reporter": "~4.2.1",
    "jest": "^23.0.0",
    "jest-preset-angular": "6.0.1",
    "karma": "~3.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~2.0.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "ngrx-store-freeze": "0.2.4",
    "prettier": "1.15.2",
    "protractor": "~5.4.0",
    "ts-node": "~7.0.0",
    "tslint": "~5.11.0",
    "typescript": "~3.2.2"
  }
}
`

**Comment by pamtbaau** [comment]
Bumped into same issue when total number of (dev)dependencies exceeds a certain threshold. Reducing the number of dependencies makes Versionlens work again. Increasing the number and it never finishes.

Vscode:
Version: 1.35.1 (user setup)
Commit: c7d83e57cd18f18026a8162d042843bda1bcf21f
Date: 2019-06-12T14:30:02.622Z
Electron: 3.1.8
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT x64 10.0.18362

**Comment by HeIp** [comment]
Solution ? @pflannery  , @fcrespo82

**Comment by marklanhamhc** [comment]
I'm getting the same issue, the plugin was working fine last week and I haven't added any more plugins, but now it just gets stuck on Analyzing dependencies... and its been like it all day.

VS Code details:
Version: 1.38.0
Commit: 3db7e09f3b61f915d03bbfa58e258d6eee843f35
Date: 2019-09-03T21:47:00.464Z
Electron: 4.2.10
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Darwin x64 18.7.0

**Comment by coder2000** [comment]
It works when I first load code and open my package.json however each time I open it after my CPU usage jumps and my system slows down and nothing happens with the plugin.

**Comment by vfonic** [comment]
This is still an issue. I just tried using Version Lens to bump up my package versions, but it's stuck forever loading.

<details>
<summary>Here's my package.json</summary>

```json
{
  "dependencies": {
    "@babel/core": "^7.7.2",
    "@babel/preset-react": "^7.7.4",
    "@hot-loader/react-dom": "^16.11.0",
    "@mapbox/mapbox-sdk": "^0.7.1",
    "@rails/webpacker": "4.2.2",
    "@turf/bbox": "4.7.3",
    "@turf/unkink-polygon": "^5.1.5",
    "accounting": "^0.4.1",
    "apollo-cache": "^1.3.2",
    "apollo-cache-inmemory": "^1.6.5",
    "apollo-client": "^2.6.8",
    "apollo-link": "^1.2.13",
    "apollo-link-error": "^1.1.12",
    "apollo-link-http": "^1.5.16",
    "apollo-utilities": "^1.3.2",
    "babel-plugin-transform-react-remove-prop-types": "^0.4.24",
    "clean-webpack-plugin": "^3.0.0",
    "cloudinary-core": "^2.8.0",
    "cloudinary-react": "^1.2.1",
    "core-js": "^3.6.0",
    "dotenv": "^8.2.0",
    "fuse.js": "^3.4.6",
    "geojson": "^0.5.0",
    "graphql": "^14.5.8",
    "graphql-tag": "^2.10.1",
    "jquery": "^3.4.1",
    "later": "^1.2.0",
    "lodash": "^4.17.15",
    "mapbox-gl": "^1.6.1",
    "moment": "^2.24.0",
    "normalize-url": "^4.5.0",
    "prop-types": "^15.7.2",
    "quill-magic-url": "https://github.com/vfonic/quill-magic-url#normalize-url-4",
    "react": "^16.12.0",
    "react-apollo": "^3.1.3",
    "react-body-classname": "^1.3.1",
    "react-content-loader": "^4.3.3",
    "react-dnd": "^9.5.1",
    "react-dnd-html5-backend": "^9.5.1",
    "react-dom": "^16.12.0",
    "react-flatpickr": "3.6.4",
    "react-geosuggest": "^2.12.1",
    "react-google-maps-loader": "^4.2.6",
    "react-helmet": "^5.2.1",
    "react-hot-loader": "^4.12.18",
    "react-infinite-scroll-component": "^4.5.3",
    "react-input-mask": "^2.0.4",
    "react-intercom": "^2.0.0-alpha.1",
    "react-mapbox-gl": "^4.8.1",
    "react-modal": "^3.11.1",
    "react-quill": "^1.3.3",
    "react-resize-observer": "^1.1.1",
    "react-router": "^5.1.2",
    "react-router-dom": "^5.1.2",
    "react-select": "^1.3.0",
    "react-stripe-elements": "^5.1.0",
    "rollbar": "^2.14.4",
    "simpl-schema": "^1.5.6",
    "slick-carousel": "^1.8.1",
    "stop-runaway-react-effects": "^1.2.1",
    "string-replace-loader": "^2.2.0",
    "toastr": "^2.1.4",
    "ts-loader": "^6.2.1",
    "typescript": "^3.7.4",
    "underscore": "^1.9.1",
    "uniforms": "~2.4.0",
    "uniforms-bridge-simple-schema-2": "~2.4.0",
    "uniforms-unstyled": "~2.4.0",
    "viewport-mercator-project": "^6.2.2",
    "whatwg-fetch": "^3.0.0"
  },
  "devDependencies": {
    "@graphql-codegen/cli": "^1.9.1",
    "@graphql-codegen/fragment-matcher": "1.9.1",
    "@graphql-codegen/introspection": "1.9.1",
    "@types/geojson": "^7946.0.7",
    "@types/lodash": "^4.14.144",
    "@types/mapbox-gl": "^0.54.4",
    "@types/react": "^16.9.9",
    "@types/react-dom": "^16.9.2",
    "@types/react-router-dom": "^5.1.0",
    "@types/webpack-env": "^1.14.1",
    "@typescript-eslint/eslint-plugin": "^2.12.0",
    "@typescript-eslint/parser": "^2.12.0",
    "eslint": "^6.8.0",
    "eslint-config-prettier": "^6.0.0",
    "eslint-plugin-import": "^2.18.2",
    "eslint-plugin-prettier": "^3.1.0",
    "eslint-plugin-react": "^7.14.3",
    "eslint-plugin-react-hooks": "^1.6.1",
    "prettier": "^1.18.2",
    "prettier-eslint": "^9.0.0",
    "prettier-eslint-cli": "^5.0.0",
    "webpack": "^4.41.2",
    "webpack-dev-server": "^3.10.1"
  },
  "private": true,
  "scripts": {
    "start": "foreman start -f Procfile.dev",
    "generate": "graphql-codegen --config codegen.yml",
    "webpack-build": "bundle exec rails assets:precompile",
    "debug-webpack-build": "NODE_ENV=production ./bin/webpack --progress --config config/webpack/production.js",
    "fix-js": "eslint --fix 'app/**/*.{ts,tsx,js,jsx}'",
    "lint-js": "eslint --quiet 'app/**/*.{ts,tsx,js,jsx}'"
  },
  "engines": {
    "node": "^12.0.0",
    "yarn": ">=1.17.3"
  }
}
```
</details>

**Comment by williamluke4** [comment]
@pflannery Is there anything that I can do to help get this resolved?

**Comment by mifas** [comment]
I'm having the same issue. 
Dependency: 47
DevDependency:  36

**Comment by linonetwo** [comment]
Hope we can use another source, for example, `registry=https://registry.npm.taobao.org/` in config, which might be faster in some region.

**Comment by pflannery** [comment]
> mine is now working but example of the package.json that was giving problems is as follows:
> `{ "name": "project-name", "version": "0.0.0", "license": "MIT", "scripts": { "ng": "ng", "start": "ng serve", "build": "ng build", "test": "ng test", "lint": "./node_modules/.bin/nx lint && ng lint", "e2e": "ng e2e", "affected:apps": "./node_modules/.bin/nx affected:apps", "affected:libs": "./node_modules/.bin/nx affected:libs", "affected:build": "./node_modules/.bin/nx affected:build", "affected:e2e": "./node_modules/.bin/nx affected:e2e", "affected:test": "./node_modules/.bin/nx affected:test", "affected:lint": "./node_modules/.bin/nx affected:lint", "affected:dep-graph": "./node_modules/.bin/nx affected:dep-graph", "format": "./node_modules/.bin/nx format:write", "format:write": "./node_modules/.bin/nx format:write", "format:check": "./node_modules/.bin/nx format:check", "update": "ng update @nrwl/schematics", "update:check": "ng update", "workspace-schematic": "./node_modules/.bin/nx workspace-schematic", "dep-graph": "./node_modules/.bin/nx dep-graph", "help": "./node_modules/.bin/nx help", "affected": "./node_modules/.bin/nx affected" }, "private": true, "dependencies": { "@angular/animations": "^7.0.0", "@angular/common": "^7.0.0", "@angular/compiler": "^7.0.0", "@angular/core": "^7.0.0", "@angular/forms": "^7.0.0", "@angular/platform-browser": "^7.0.0", "@angular/platform-browser-dynamic": "^7.0.0", "@angular/router": "^7.0.0", "@auth0/angular-jwt": "^2.1.0", "@coreui/angular": "^2.2.4", "@coreui/coreui": "^2.1.6", "@coreui/coreui-plugin-chartjs-custom-tooltips": "^1.2.0", "@coreui/icons": "^0.3.0", "@ng-idle/core": "^6.0.0-beta.3", "@ng-idle/keepalive": "^6.0.0-beta.3", "@ngrx/effects": "7.2.0", "@ngrx/entity": "6.1.0", "@ngrx/router-store": "7.2.0", "@ngrx/store": "7.2.0", "@ngx-translate/core": "^11.0.1", "@ngx-translate/http-loader": "^4.0.0", "@nrwl/nx": "7.6.2", "@sweetalert2/ngx-sweetalert2": "^5.0.1", "angular-pipes": "^9.0.2", "angular2-moment": "^1.9.0", "angular2-toaster": "^7.0.0", "bootstrap": "^4.3.1", "chart.js": "^2.8.0", "core-js": "^2.5.4", "flag-icon-css": "^3.3.0", "font-awesome": "^4.7.0", "google-libphonenumber": "^3.2.2", "libphonenumber-js": "^1.7.15", "moment": "^2.24.0", "mutationobserver-shim": "^0.3.3", "ng-bootstrap-form-validation": "^4.0.0", "ng-spin-kit": "^5.1.1", "ng2-charts": "^2.2.2", "ngx-bootstrap": "^4.0.1", "ngx-loading": "^3.0.1", "ngx-pagination": "^3.2.1", "ngx-perfect-scrollbar": "^7.2.0", "ngx-permissions": "^6.0.5", "ngx-uploader": "^7.1.0", "rxjs": "~6.4.0", "simple-line-icons": "^2.4.1", "sweetalert2": "^8.8.7", "tslib": "^1.9.0", "v8n": "^1.2.3", "zone.js": "^0.8.26" }, "devDependencies": { "@angular-devkit/build-angular": "~0.13.1", "@angular/cli": "7.3.1", "@angular/compiler-cli": "^7.0.0", "@angular/language-service": "^7.0.0", "@ngrx/schematics": "7.2.0", "@ngrx/store-devtools": "7.2.0", "@nrwl/builders": "7.6.2", "@nrwl/schematics": "7.6.2", "@types/jasmine": "~2.8.6", "@types/jasminewd2": "~2.0.3", "@types/jest": "^23.0.0", "@types/jquery": "3.3.6", "@types/node": "~8.9.4", "codelyzer": "~4.5.0", "cypress": "^3.1.0", "dotenv": "6.2.0", "jasmine-core": "~2.99.1", "jasmine-marbles": "0.4.0", "jasmine-spec-reporter": "~4.2.1", "jest": "^23.0.0", "jest-preset-angular": "6.0.1", "karma": "~3.0.0", "karma-chrome-launcher": "~2.2.0", "karma-coverage-istanbul-reporter": "~2.0.1", "karma-jasmine": "~1.1.0", "karma-jasmine-html-reporter": "^0.2.2", "ngrx-store-freeze": "0.2.4", "prettier": "1.15.2", "protractor": "~5.4.0", "ts-node": "~7.0.0", "tslint": "~5.11.0", "typescript": "~3.2.2" } } `

I find that example you provided works fast for me (even if I dont format the json)

![image](https://user-images.githubusercontent.com/1727302/80922963-72cf4680-8d78-11ea-9033-8ea011929d92.png)

**Comment by pflannery** [comment]
@vfonic
> This is still an issue. I just tried using Version Lens to bump up my package versions, but it's stuck forever loading.
> 
> Here's my package.json

I cant seem to repeat this issue with the package.json you provided, it works fast for me...if it gets stuck again can you open the dev tools console check for any errors. thanks

**Comment by pflannery** [comment]
@josh-degraw the npm package version has been updated since you reported this issue. Do u still get this problem?

**Comment by pamtbaau** [comment]
Package with ~50 dependencies is working now. I say 'now' because it has been on and off before...

**Comment by graycrow** [comment]
Fresh install of the plugin, never installed before. VS Code 1.44.2, Manjaro Linux. Have a few projects with 60, 101 and 114 dependencies. Plugin doesn't work for all of them, icon is spinning, no network activity, just CPU load. Tried to clean a npm cache - no change, maybe because I'm using Yarn on those projects.

**Comment by pflannery** [comment]
Found a related issue in #190 where npm view was caching the pluging. I have rolled npm back to a known working version. I've also added better error handling. I believe that these issues should now be resolved in 0.26.2

@graycrow try 0.26.2...the fortawesome packages are working for me (at least the ones that exist)

**Comment by graycrow** [comment]
@pflannery, with 0.26.2 it works everywhere, thank you!

**Comment by josh-degraw** [comment]
@pflannery I am no longer working on the original codebase I was working on when I raised this, but it seems to be working with the large list of dependencies I just tested with! Seems like it should be fixed!

@graycrow I wonder if that is related to my original issue, I _think_ I was referencing one or more of those @FortAwesome packages in the project when I raised this...

But by and large the root issue seems to be fixed! I'll go ahead and close this issue.
